lib/utils/helpers.coffee

```diff
@@ -93,10 +93,25 @@ exports.osProgressHandler = (step) ->
 	return rindle.wait(step)
 
 exports.getAppInfo = (application) ->
+	# Check for an app of the form `user/application`, and send
+	# this off to a special handler (before importing any modules)
+	if (match = /(\w+)\/(\w+)/.exec(application))
+		return getAppInfoWithUser(match[1], match[2])
+
+	resin = require('resin-sdk-preconfigured')
+	return getAppInfoFromPromise(resin.models.application.get(application))
+
+exports.getAppInfoWithUser = getAppInfoWithUser = (username, appName) ->
+	resin = require('resin-sdk-preconfigured')
+	return getAppInfoFromPromise(resin.models.application.getAppWithOwner(appName, username))
+
+# Helper function to setup application info from a promise that resolves
+# to an application
+getAppInfoFromPromise = (appPromise) ->
 	resin = require('resin-sdk-preconfigured')
 	_ = require('lodash')
 	Promise.join(
-		resin.models.application.get(application),
+		appPromise,
```

Comment:
Much, much better :ok_hand: definitely refactoring this.

Re: the exporting of `getAppInfoWithUser`, I was exporting it simply because it may be useful. I think if we do want it to be exported, your method is also nicer, but if we don't want it to be exported because it won't be used anywhere at first, that's also fine too. I think given the subtlety of your method, we might as well export it like that.